src/Hamiltonians/HubbardMomSpace.jl

```diff
@@ -0,0 +1,639 @@
+"""
+    dispersion_MomSpace(ks, geometry, t)
+    Dispersion relation for a given set of k values, lattice geometry and hopping strengths t. Returns
+        ``-2(\\sum_{\\bar{k}} \\Re(t_{\\bar{k}}) \\cos(k_{\\bar{k}}) + \\Im(t_{\\bar{k}}) \\sin(k_{\\bar{k}}))``
+    where ``\\bar{k}`` runs over all dimensions of the lattice.
+"""
+function dispersion_MomSpace(ks::SVector{D}, geometry::CubicGrid{D, S}, t::SMatrix) where {D,S}
+    # Calculate the dispersion relation for a given set of k values and hopping strength t.
+    C,_ = size(t)
+    M = prod(S)
+    kes_mat = zeros(Float64,C,M)
+    ks_mat = zeros(Float64,D,M)
+    phase = atan.(imag.(t)./real.(t))
+    for j in 1:C
+        for i in 1:M
+            mom_val = value_of_mom_mode(M-i+1, ks, geometry)
+            kes_mat[j,i] = convert(Float64,hub_dis_Mom_Space(t[j,:], mom_val)[1])
+            ks_mat[:,i] = mom_val + phase[j,:]
+        end
+    end
+    return SMatrix{C,M,Float64}(kes_mat), SMatrix{D,M,Float64}(ks_mat)
+end
+function value_of_mom_mode(add_index::Int, ks::SVector{D}, geometry::CubicGrid{D}) where {D}
+    mom_mode = reverse(geometry[add_index])
+    return [ks[i][mode] for (i, mode) in enumerate(mom_mode)]
+end
+
+function hub_dis_Mom_Space(t::SVector, k::Vector)
+    # Calculate the dispersion relation for a given k value and hopping strength t.
+    return -2 * (real.(t') * cos.(k) - imag.(t') * sin.(k))
+end
+
+"""
+    _mom_hopping(kes, address)
+
+    Calculate the hopping term for a single-component or multi-component Fock state address in momentum space.
+"""
+@inline function _mom_hopping(kes::SMatrix{C}, address::CompositeFS{C}) where {C}
+    # Calculate the hopping term for a single component.
+    comp = address.components
+    onproduct = 0.0
+    for i in 1:C
+        onproduct += dot(kes[i,:], OccupiedModeMap(comp[i]))
+    end
+    return onproduct
+end
+
+@inline _mom_hopping(kes::SMatrix{1}, address::SingleComponentFockAddress) = dot(kes[1,:], OccupiedModeMap(address))
+
+"""
+    mom_transfer_MomSpace(add, chosen, map, g; fold=true)
+    mom_transfer_MomSpace(add1, add2, chosen, map1, map2, g; fold=true)
+
+Get the momentum transfer for a given excitation in a same or two different components of a
+multi-component Fock state address in momentum space, i.e., `add` or between `add1` and `add2`. 
+`map`, `map1` and `map2` are the occupied mode maps for the relevant components of the 
+multi-component Fock state. `chosen` is an integer that determines which excitation and `g` is
+a geometry of the lattice.
+
+"""
+@inline function mom_transfer_MomSpace(
```

Comment:
```suggestion
@inline function mom_transfer_mom_space(
```
same here (and elsewhere).

The name of this function is a bit weird. Could this be merged with/replace `momentum_transfer_excitation`?